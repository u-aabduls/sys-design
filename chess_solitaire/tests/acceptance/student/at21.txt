-- ACCEPTANCE TEST #21: VIOLATION CASE: moves - slot_occupied

-- Setup some chess pieces for the game
setup_chess (K,1,3)
setup_chess (Q,2,4)
setup_chess (B,2,3)
setup_chess (R,4,1)

-- start game
start_game

-- check moves for (1,1), an unoccupied slot
moves(1,1)

-- check moves for (1,3), an occupied slot
moves(1,3)

-- check moves for (2,2), an unoccupied slot
moves(2,2)

-- check moves for (2,4), an occupied slot
moves(2,4)

-- check moves for (3,3), an unoccupied slot
moves(3,3)

-- check moves for (2,3), an occupied slot
moves(2,3)

-- check moves for (4,4), an unoccupied slot
moves(4,4)

-- check moves for (4,1), an occupied slot
moves(4,1)

-- check moves for (1,4), an unoccupied slot
moves(1,4)