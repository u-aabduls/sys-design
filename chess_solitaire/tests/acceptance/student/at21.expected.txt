  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(K,1,3)
  # of chess pieces on board: 1
  Game being Setup...
  ..K.
  ....
  ....
  ....
->setup_chess(Q,2,4)
  # of chess pieces on board: 2
  Game being Setup...
  ..K.
  ...Q
  ....
  ....
->setup_chess(B,2,3)
  # of chess pieces on board: 3
  Game being Setup...
  ..K.
  ..BQ
  ....
  ....
->setup_chess(R,4,1)
  # of chess pieces on board: 4
  Game being Setup...
  ..K.
  ..BQ
  ....
  R...
->start_game
  # of chess pieces on board: 4
  Game In Progress...
  ..K.
  ..BQ
  ....
  R...
->moves(1,1)
  # of chess pieces on board: 4
  Error: Slot @ (1, 1) not occupied
  ..K.
  ..BQ
  ....
  R...
->moves(1,3)
  # of chess pieces on board: 4
  Game In Progress...
  .+K+
  .+++
  ....
  ....
->moves(2,2)
  # of chess pieces on board: 4
  Error: Slot @ (2, 2) not occupied
  ..K.
  ..BQ
  ....
  R...
->moves(2,4)
  # of chess pieces on board: 4
  Game In Progress...
  ..++
  +++Q
  ..++
  .+.+
->moves(3,3)
  # of chess pieces on board: 4
  Error: Slot @ (3, 3) not occupied
  ..K.
  ..BQ
  ....
  R...
->moves(2,3)
  # of chess pieces on board: 4
  Game In Progress...
  .+.+
  ..B.
  .+.+
  +...
->moves(4,4)
  # of chess pieces on board: 4
  Error: Slot @ (4, 4) not occupied
  ..K.
  ..BQ
  ....
  R...
->moves(4,1)
  # of chess pieces on board: 4
  Game In Progress...
  +...
  +...
  +...
  R+++
->moves(1,4)
  # of chess pieces on board: 4
  Error: Slot @ (1, 4) not occupied
  ..K.
  ..BQ
  ....
  R...
